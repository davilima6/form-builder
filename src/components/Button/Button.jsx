// @flow
import * as React from 'react';
import styled, { css } from 'styled-components';
import { errorMessages } from '../../utils/validators';

type Props = {
  floated?: string,
  onClick?: Function,
  primary?: boolean,
  disabled?: boolean,
  title: string,
  type?: string,
};

const StyledButton = styled.button`
  background-color: #fff;
  border: 1px solid var(--lite-grey);
  border-radius: 3px;
  font-size: 1em;
  margin-right: ${props => (props.floated === 'left' ? 'auto' : '0')};
  order: ${props => (props.floated === 'left' ? '0' : '1')};
  padding: 0.5em 2em;

  :active {
    border-color: #000;
  }

  ${props => props.primary
    && css`
      background-color: var(--dark-green);
      border-color: var(--dark-green);
      color: #fff;
      cursor: pointer;
      font-weight: 500;

      :hover {
        background-color: var(--lite-green);
        color: #000;
      }
    `};

  ${props => props.disabled
    && css`
      background-color: var(--lite-grey);
      border-color: var(--dark-grey);
      color: #000;
      cursor: not-allowed;

      :hover {
        background-color: var(--lite-grey);
      }
    `};
`;

function Button({
  floated, onClick, primary, disabled, title, type,
}: Props) {
  return (
    <StyledButton
      type={type}
      floated={floated}
      title={disabled ? errorMessages.isPristine() : title}
      primary={primary}
      disabled={disabled}
      onClick={onClick}
    >
      {title}
    </StyledButton>
  );
}

Button.defaultProps = {
  floated: null,
  onClick: () => {},
  primary: false,
  disabled: false,
  type: 'button',
};

export default Button;
