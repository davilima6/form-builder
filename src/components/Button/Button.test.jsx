import React from 'react';
import { cleanup, render } from 'react-testing-library';
import 'jest-dom/extend-expect';
// import 'jest-styled-components';
import Button from './Button';

describe('Button', () => {
  let component;

  beforeEach(() => {
    cleanup();
    component = render(<Button />);
  });

  test('renders without crashing', () => expect(component).toBeDefined());

  test('matches snapshot', () => {
    const { asFragment } = component;

    expect(asFragment()).toMatchSnapshot();
  });

  test('matches snapshot for disabled state', () => {
    component = render(<Button disabled />);
    const { asFragment } = component;

    expect(asFragment()).toMatchSnapshot();
  });
});
