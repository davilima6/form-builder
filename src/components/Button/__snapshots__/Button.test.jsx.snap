// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bdVaJa hjNKPk"
    type="button"
  />
</DocumentFragment>
`;

exports[`Button matches snapshot for disabled state 1`] = `
<DocumentFragment>
  <button
    class="sc-bdVaJa eNYMjX"
    disabled=""
    title="Form has not been filled out yet so it cannot be submitted"
    type="button"
  />
</DocumentFragment>
`;
